gen3/bin/workon.sh

```diff
@@ -0,0 +1,217 @@
+#
+# Helper script for 'gen3 workon' - see ../README.md and ../gen3setup.sh
+#
+
+if [[ ! -f "$GEN3_HOME/gen3/lib/common.sh" ]]; then
+  echo "ERROR: no $GEN3_HOME/gen3/lib/common.sh"
+  exit 1
+fi
+
+help() {
+  cat - <<EOM
+  Use: gen3 workon aws-profile vpc-name
+     Prepares a local workspace to run terraform and other devops tools.
+EOM
+  return 0
+}
+
+source "$GEN3_HOME/gen3/lib/common.sh"
+
+#
+# Create any missing files
+#
+mkdir -p -m 0700 "$GEN3_WORKDIR/backups"
+
+#
+# aws_provider.tfvars - this has the secret keys that
+# the terraform aws provider wants:
+#     https://www.terraform.io/docs/providers/aws/
+#
+if [[ ! -f "$GEN3_WORKDIR/aws_provider.tfvars" ]]; then
+  echo "Creating $GEN3_WORKDIR/aws_provider.tfvars"
+  cat - > "$GEN3_WORKDIR/aws_provider.tfvars" <<EOM
+aws_access_key = "$(aws configure get "$GEN3_PROFILE.aws_access_key_id")"
+aws_secret_key = "$(aws configure get "$GEN3_PROFILE.aws_secret_access_key")"
+aws_region = "$(aws configure get "$GEN3_PROFILE.region")"
+EOM
+fi
+
+#
+# aws_backend.tfvars - this has the secret keys that
+# the terraform S3 backend wants:
+#     https://www.terraform.io/docs/backends/types/s3.html
+#
+if [[ ! -f "$GEN3_WORKDIR/aws_backend.tfvars" ]]; then
+  echo "Creating $GEN3_WORKDIR/aws_backend.tfvars"
+  cat - > "$GEN3_WORKDIR/aws_backend.tfvars" <<EOM
+access_key = "$(aws configure get "$GEN3_PROFILE.aws_access_key_id")"
+secret_key = "$(aws configure get "$GEN3_PROFILE.aws_secret_access_key")"
+region = "$(aws configure get "$GEN3_PROFILE.region")"
+EOM
+fi
+
+
+#
+# Sync the given file with S3.
+# Note that 'workon' only every copies from S3 to local,
+# and only if a local copy does not already exist.
+# See 'gen3 refresh' to pull down latest files from s3.
+# We copy the local up to S3 at 'apply' time.
+#
+refreshFromS3() {
+  local fileName
+  local filePath
+  fileName=$1
+  if [[ -z $fileName ]]; then
+    return 1
+  fi
+  filePath="${GEN3_WORKDIR}/$fileName"
+  if [[ -f $filePath ]]; then
+    echo "Ignoring S3 refresh for file that already exists: $fileName"
+    return 1
+  fi
+  s3Path="s3://${S3_TERRAFORM}/${GEN3_VPC}/${fileName}"
+  aws s3 cp "$s3Path" "$filePath" > /dev/null 2>&1
+  if [[ ! -f "$filePath" ]]; then
+    echo "No data at $s3Path"
+    return 1
+  fi
+  return 0
+}
+
+#
+# Let helper generates a random string of alphanumeric characters of length $1.
+#
+function random_alphanumeric() {
+    base64 /dev/urandom | tr -dc 'a-zA-Z0-9' | head -c $1
+}
+
+
+#
+# Generate an initial backend.tfvars file with intelligent defaults
+# where possible.
+#
+backend.tfvars() {
+  cat - <<EOM
+bucket = "cdis-terraform-state"
```

Comment:
is bucket this owned by cdistest account?
So if the aws creds provided is for a commons  under a different aws account it won't work?